### DISCUSSION TITLE
Helpful tip when uploading large files

### DISCUSSION THREAD

**Discussion by tnylea** [livewire/livewire#4438] [2. Show and tell]
Hey guys,

Just wanted to share a quick tip that I found useful when uploading large files. When uploading large files you may want to move the file from temporary storage to the new location found in this discussion: https://github.com/livewire/livewire/discussions/4003.

The other problem is that you may want to validate the filesize before allowing the user to upload the temporary file. By adding `event.stopImmediatePropagation` to your input, it will help with this. Here is an example:

```
<input type="file" onchange="if(this.files[0].size > '100000000'){ event.stopImmediatePropagation(); alert('File uploads cannot be larger than 100MB.') }" wire:model="file"/>
```

In the example above ☝️ the user will be given an alert if they try and upload a file larger than `100MB`. Of course, the user could always modify the DOM and change that rule; but if you have the same filesize validation in your Livewire Controller it will prevent them from permanently storing the large file.

This will prevent user frustration in the following scenario: they go to upload a large file `> 100MB` and after the upload finishes and they click to save, it will prompt them and tell them the filesize is too large. That's why it's good to have this check in place even before they upload the temporary file.

Hope this helps someone else out in the future 😉 

Cheers 🍻

**Comment by joshhanley** [comment]
@tnylea thanks for sharing! I've moved this to the show and tell category 🙂

**Comment by adham-shafik** [comment]
Unfortunately  `event.stopImmediatePropagation();` didn't prevent livewire from uploading the file.
Instead i made a work around, to make Alpine `$wire.set('size_exceeded',true)` if size is larger than 10 MB, and inside Livewire's component, i used 
`updatingFile(){
    if($size_exceede){
         $this->file=null;
         $this->validate();
   }
   $this->size_exceeded=false;
}`
